pkg/client/restclient/request.go

```diff
@@ -821,9 +821,16 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		}
 
 		done := func() bool {
-			// ensure the response body is closed before we reconnect, so that we reuse the same
-			// TCP connection
-			defer resp.Body.Close()
+			// Ensure the response body is fully read and closed
+			// before we reconnect, so that we reuse the same TCP
+			// connection.
+			defer func() {
+				const maxBodySlurpSize = 2 << 10
```

Comment:
Would recommend instead:

```
io.Copy(ioutil.Discard, ioutil.LimitedReader{R: resp.Body, N: maxBodySlurpSize})
```

Do you need to check ContentLength -1 in this case?  OR just check for body nil.